IllegalStateException in WatchServiceImpl at shutdown

In build 3320 I see this exception at shutdown. I know this area is still under some development, but thought it best to log an issue.

```
2023-02-19 09:18:00.555 [ERROR] [re.internal.service.WatchServiceImpl] - bundle org.openhab.core:4.0.0.202302181816 (151)[org.openhab.core.internal.service.WatchServiceImpl(257)] : The deactivate method has thrown an exception
java.lang.IllegalStateException: The service has been unregistered {org.openhab.core.service.WatchService}={service.id=413, service.bundleid=151, service.scope=singleton, watchservice.name=configWatcher}
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.unregister(ServiceRegistrationImpl.java:224) ~[org.eclipse.osgi-3.17.200.jar:?]
	at org.openhab.core.internal.service.WatchServiceImpl.closeWatcherAndUnregister(WatchServiceImpl.java:153) ~[?:?]
	at org.openhab.core.internal.service.WatchServiceImpl.deactivate(WatchServiceImpl.java:130) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invokeMethod(BaseMethod.java:244) ~[bundleFile:?]
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.access$500(BaseMethod.java:41) ~[bundleFile:?]
	at org.apache.felix.scr.impl.inject.methods.BaseMethod$Resolved.invoke(BaseMethod.java:685) ~[bundleFile:?]
	at org.apache.felix.scr.impl.inject.methods.BaseMethod.invoke(BaseMethod.java:529) [bundleFile:?]
	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:318) [bundleFile:?]
	at org.apache.felix.scr.impl.inject.methods.ActivateMethod.invoke(ActivateMethod.java:308) [bundleFile:?]
	at org.apache.felix.scr.impl.manager.SingleComponentManager.disposeImplementationObject(SingleComponentManager.java:421) [bundleFile:?]
	at org.apache.felix.scr.impl.manager.SingleComponentManager.deleteComponent(SingleComponentManager.java:165) [bundleFile:?]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.doDeactivate(AbstractComponentManager.java:853) [bundleFile:?]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.deactivateInternal(AbstractComponentManager.java:825) [bundleFile:?]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.dispose(AbstractComponentManager.java:589) [bundleFile:?]
	at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.configurationDeleted(ConfigurableComponentHolder.java:332) [bundleFile:?]
	at org.apache.felix.scr.impl.manager.RegionConfigurationSupport.configurationEvent(RegionConfigurationSupport.java:310) [bundleFile:?]
	at org.apache.felix.scr.impl.manager.RegionConfigurationSupport$2.configurationEvent(RegionConfigurationSupport.java:115) [bundleFile:?]
	at org.apache.felix.cm.impl.ConfigurationManager$FireConfigurationEvent.sendEvent(ConfigurationManager.java:1721) [bundleFile:?]
	at org.apache.felix.cm.impl.ConfigurationManager$FireConfigurationEvent.run(ConfigurationManager.java:1662) [bundleFile:?]
	at org.apache.felix.cm.impl.UpdateThread.run0(UpdateThread.java:122) [bundleFile:?]
	at org.apache.felix.cm.impl.UpdateThread.run(UpdateThread.java:84) [bundleFile:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
```

The only unusual thing I see in the log before the exception is this. But it's 5 sec before the exception is thrown.
```
2023-02-19 13:42:34.544 [WARN ] [core.thing.internal.ThingManagerImpl] - Disposing handler for thing 'miio:generic:03D9D158' takes more than 5000ms.
2023-02-19 13:42:39.603 [ERROR] [re.internal.service.WatchServiceImpl] - bundle org.openhab.core:4.0.0.202302181816 (151)[org.openhab.core.internal.service.WatchServiceImpl(44)] : The deactivate method has thrown an exception
java.lang.IllegalStateException: The service has been unregistered {org.openhab.core.service.WatchService}={service.id=215, service.bundleid=151, service.scope=singleton, watchservice.name=configWatcher}
```

--- a/bundles/org.openhab.core/src/main/java/org/openhab/core/internal/service/WatchServiceImpl.java
+++ b/bundles/org.openhab.core/src/main/java/org/openhab/core/internal/service/WatchServiceImpl.java
@@ -149,7 +149,7 @@ public class WatchServiceImpl implements WatchService, DirectoryChangeListener {
         }
 
         ServiceRegistration<?> localReg = this.reg;
-        if (localReg != null) {
+        if (localReg != null && bundleContext.getService(localReg.getReference()) != null) {
             localReg.unregister();
             this.reg = null;
         }

